tests/GraphBLAS-sharp.Tests/OperationsTests/EWiseAddTests.fs

```diff
@@ -0,0 +1,237 @@
+            (fun x -> abs x < Accuracy.medium.absolute)
+
+    static member BoolType() =
+        Arb.fromGen <| Generators.pairOfSparseMatricesGenerator
+            Arb.generate<bool>
+            false
+            ((=) false)
+
+let createMatrix<'a when 'a : struct and 'a : equality> matrixFormat args =
+    match matrixFormat with
+    | CSR ->
+        Activator.CreateInstanceGeneric<CSRMatrix<_>>(
+            Array.singleton typeof<'a>, args
+        )
+        |> unbox<CSRMatrix<'a>>
+        :> Matrix<'a>
+    | COO ->
+        Activator.CreateInstanceGeneric<COOMatrix<_>>(
+            Array.singleton typeof<'a>, args
+        )
+        |> unbox<COOMatrix<'a>>
+        :> Matrix<'a>
+
+let logger = Log.create "Sample"
+
+let correctnessOnNumbers<'a when 'a : struct and 'a : equality>
+    (sum: 'a -> 'a -> 'a)
+    (diff: 'a -> 'a -> 'a)
+    (isZero: 'a -> bool)
+    (semiring: Semiring<'a>)
+    (case: OperationCase)
+    (matrixA: 'a[,], matrixB: 'a[,]) =
+
+    let eWiseAddNaive (matrixA: 'a[,]) (matrixB: 'a[,]) =
+        let left = matrixA |> Seq.cast<'a>
+        let right = matrixB |> Seq.cast<'a>
+
+        (left, right)
+        ||> Seq.map2
+            (fun x y ->
+                if isZero x && isZero y then None
+                else Some <| sum x y
+            )
+        |> Seq.choose id
+
+    let eWiseAddGB (matrixA: 'a[,]) (matrixB: 'a[,]) =
+        try
+            let left = createMatrix<'a> case.MatrixCase [|matrixA; isZero|]
+            let right = createMatrix<'a> case.MatrixCase [|matrixB; isZero|]
+
+            logger.debug (
+                eventX "Left matrix is \n{matrix}"
+                >> setField "matrix" left
+            )
+
+            logger.debug (
+                eventX "Right matrix is \n{matrix}"
+                >> setField "matrix" right
+            )
+
+            opencl {
+                let! result = left.EWiseAdd right None semiring
+                let! tuples = result.GetTuples()
+                return! tuples.ToHost()
+            }
+            |> oclContext.RunSync
+            |> (fun tuples -> tuples.Values)
+            |> Seq.ofArray
+
+        finally
+            oclContext.Provider.CloseAllBuffers()
+
+    let expected = eWiseAddNaive matrixA matrixB
+    let actual = eWiseAddGB matrixA matrixB
+
+    logger.debug (
+        eventX "Expected result is {matrix}"
+        >> setField "matrix" (sprintf "%A" <| List.ofSeq expected)
+    )
+
+    logger.debug (
+        eventX "Actual result is {matrix}"
+        >> setField "matrix" (sprintf "%A" <| List.ofSeq actual)
+    )
+
+    "Length of expected and result seq should be equal"
+    |> Expect.hasLength actual (Seq.length expected)
+
+    let difference =
+        (expected, actual)
+        ||> Seq.map2 diff
+
+    "There should be no difference between expected and received values"
+    |> Expect.all difference isZero
+
+let correctnessOnBool (case: OperationCase) (matrixA: bool[,], matrixB: bool[,]) =
```

Comment:
Why it can't be generalized? It seems to be particular case of `correctnessOnNumbers`